tests/tree/test_tree.cxx

```diff
@@ -44,27 +44,25 @@ const std::string XMLDATA_BAD =
 }
 
 
-BOOST_AUTO_TEST_SUITE( tree )
-
 /*
  * tests that a good XML is parsed correctly.
  */
 
-BOOST_AUTO_TEST_CASE( good_xml_1 )
+TEST_CASE_METHOD( SrcdirConfig, "tree/good_xml_1", "[tree]" )
 {
     xml::tree_parser parser(test_file_path("tree/data/good.xml").c_str());
-    BOOST_CHECK( parser );
+    CHECK( !!parser );
 }
 
 
 /*
  * tests that a good XML is parsed correctly.
  */
 
-BOOST_AUTO_TEST_CASE( good_xml_2 )
+TEST_CASE_METHOD( SrcdirConfig, "tree/good_xml_2", "[tree]" )
 {
     xml::tree_parser parser(test_file_path("tree/data/good.xml").c_str(), false);
-    BOOST_CHECK( parser );
+    CHECK( !!parser );
```

Comment:
Hmm, maybe 
```suggestion
    CHECK( !parser == false );
```
Not sure if it's really better though... Perhaps a comment saying "test overloaded tree_parser::operator!()" could be helpful here. OTOH I see this is repeated many times, and we don't want to repeat the comment neither.

Maybe have a helper `inline bool is_parser_valid(tree_parser& parser) { return !!parser; }`?